### ISSUE TITLE
ERC20Wrapper `underlying` variable is public

### ISSUE THREAD

**Issue by ernestognw** [OpenZeppelin/openzeppelin-contracts#4026]
**📝 Details**

For ERC20Wrapper, the Upgradeable version's [`underlying` variable is public ](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/token/ERC20/extensions/ERC20WrapperUpgradeable.sol#L22) and the immutable is removed.

We don't want that since child contracts should not be able to override it.

It's then needed to replace the `public` visibility of the `underlying` variable [in the ERC20Wrapper contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/a035b235b4f2c9af4ba88edc4447f02e37f8d124/contracts/token/ERC20/extensions/ERC20Wrapper.sol#L19) and create a public view function that replaces its access.

**Comment by kimanikelly** [comment]
@ernestognw @frangio 

Are there any other details I'm missing to complete this task, if so please let me know, so I can make those changes and submit a  PR, thanks.

Removed the `public` visibility from `IERC20 immutable _underlying;` and refactored it to include the `_` due to linting errors.
<img width="337" alt="Screen Shot 2023-02-04 at 9 35 43 AM" src="https://user-images.githubusercontent.com/42893948/216773231-ce0ec8b9-11c2-46c2-b077-6c48a27d6cf2.png">

Implemented the `underlying() public view` function to return the value of the `_underlying` variable
<img width="755" alt="Screen Shot 2023-02-04 at 9 35 20 AM" src="https://user-images.githubusercontent.com/42893948/216773211-ccc834b2-aca1-4498-beba-dac6c5b7216e.png">


A test for `underlying` was already included in `test/token/ERC20/extensions/ERC20Wrapper.test.js` but after the refactor the call targets the new `public view` function instead of the previous `public underlying` variable
<img width="809" alt="Screen Shot 2023-02-04 at 9 42 34 AM" src="https://user-images.githubusercontent.com/42893948/216773513-c4718530-ea3d-43cc-b16a-a0d3c8997314.png">

**Comment by ernestognw** [comment]
@kimanikelly Thanks!

Feel free to open a PR for this, it doesn't require major test changes since it's already covered as you mention, and maybe only a Changeset entry.

I think we all agree on the implementation of this so we can add comments to your PR

**Comment by kimanikelly** [comment]
@ernestognw I went ahead and created the PR https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4029. All tests/checks have passed and are currently pending on a reviewer, thanks
